lib/phoenix/endpoint/supervisor.ex

```diff
@@ -135,6 +137,27 @@ defmodule Phoenix.Endpoint.Supervisor do
     end
   end
 
+  defp check_origin_or_csrf_checked!(otp_app, endpoint, socket_opts) do
+    endpoint_check_origin = config(otp_app, endpoint)[:check_origin]
+
+    for {transport, transport_opts} <- socket_opts do
+      check_origin =
+        if is_boolean(transport_opts[:check_origin]) do
+          transport_opts[:check_origin]
+        else
+          endpoint_check_origin
+        end
+
+      check_csrf = transport_opts[:check_csrf]
+
+      if check_origin == false and check_csrf == false do
+        raise ArgumentError,
+              "One of :check_origin and :check_csrf must be set to non-false value for " <>
+                "transport #{inspect(transport)}"
+      end
```

Comment:
Can you please add a test? Look at endpoint_test and you should be able to verify that `Endpoint.start_link` fails if these are disabled.

Reply:
Thanks for your feedback! Done in https://github.com/phoenixframework/phoenix/pull/5952/commits/7018509e66a5b3e7bded506b46eebf53463e124b:
- in tests we check both when `check_origin` is set as endpoint option or transport option
- we do the security check even if the transport doesn't have a `child_spec/1`